src/Nethermind/Nethermind.Specs/ChainSpecStyle/IChainSpecParametersProvider.cs

```diff
@@ -0,0 +1,13 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Specs.ChainSpecStyle;
+
+using System.Collections.Generic;
+
+public interface IChainSpecParametersProvider
+{
+    string SealEngineType { get; }
+    ICollection<IChainSpecEngineParameters> AllChainSpecParameters { get; }
```

Comment:
If we don't intend to allow modification (which I think we don't) prefer `IEnumerable` to `ICollection`